patches/@eth-optimism+contracts+0.1.11.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/node_modules/@eth-optimism/contracts/build/contracts/OVM/bridge/tokens/Abs_L2DepositedToken.sol b/node_modules/@eth-optimism/contracts/build/contracts/OVM/bridge/tokens/Abs_L2DepositedToken.sol
+index dd56f40..10e5dc9 100644
+--- a/node_modules/@eth-optimism/contracts/build/contracts/OVM/bridge/tokens/Abs_L2DepositedToken.sol
++++ b/node_modules/@eth-optimism/contracts/build/contracts/OVM/bridge/tokens/Abs_L2DepositedToken.sol
+@@ -45,6 +45,7 @@ abstract contract Abs_L2DepositedToken is iOVM_L2DepositedToken, OVM_CrossDomain
+     constructor(
+         address _l2CrossDomainMessenger
+     )
++        internal
+         OVM_CrossDomainEnabled(_l2CrossDomainMessenger)
+     {}
+ 
+diff --git a/node_modules/@eth-optimism/contracts/build/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol b/node_modules/@eth-optimism/contracts/build/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol
+index e329f90..a2525bf 100644
+--- a/node_modules/@eth-optimism/contracts/build/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol
++++ b/node_modules/@eth-optimism/contracts/build/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol
+@@ -19,7 +19,9 @@ contract OVM_CrossDomainEnabled {
+      ***************/    
+     constructor(
+         address _messenger
+-    ) {
++    )
++        public
++    {
+         messenger = _messenger;
+     }
+ 
+diff --git a/node_modules/@eth-optimism/contracts/build/src/contract-dumps.js b/node_modules/@eth-optimism/contracts/build/src/contract-dumps.js
+index cdfda5e..8c29208 100644
+--- a/node_modules/@eth-optimism/contracts/build/src/contract-dumps.js
```

Comment:
The rest are dependency conflicts between Belt's Ethers v4 & Optimism's Ethers v5 🙈

Reply:
The problem is that we use belt to create abis for our gethwrapper generation. Once we figure out a way to get [this PR](https://github.com/smartcontractkit/chainlink/pull/4010) through, it should pave the way for hardhat.

Reply:
@alexroan how is that PR going to work in the Optimism context, where we have a different (modified) solc compiler, a reimplementation of the contracts, and slightly modified ABI (still mostly backward compatible for now)?